VAMobile/src/screens/SyncScreen/SyncScreen.tsx

```diff
@@ -47,15 +47,26 @@ const SyncScreen: FC<SyncScreenProps> = () => {
 
   useEffect(() => {
     if (loggedIn) {
-      if (!personalInformationLoaded) {
+      if (!personalInformationLoaded && !personalInformationLoading) {
         dispatch(getProfileInfo())
-      } else if (authorizedServicesLoaded && militaryInfoAuthorization && !militaryHistoryLoaded) {
+      } else if (authorizedServicesLoaded && militaryInfoAuthorization && !militaryHistoryLoaded && !militaryHistoryLoading) {
         dispatch(getServiceHistory())
-      } else if (!disabilityRatingLoaded) {
+      } else if (!disabilityRatingLoaded && !disabilityRatingLoading) {
         dispatch(getDisabilityRating())
       }
     }
-  }, [dispatch, loggedIn, personalInformationLoaded, militaryInfoAuthorization, authorizedServicesLoaded, disabilityRatingLoaded, militaryHistoryLoaded])
+  }, [
+    dispatch,
+    loggedIn,
+    personalInformationLoaded,
+    personalInformationLoading,
+    militaryInfoAuthorization,
+    authorizedServicesLoaded,
+    disabilityRatingLoaded,
+    disabilityRatingLoading,
+    militaryHistoryLoaded,
+    militaryHistoryLoading,
+  ])
```

Comment:
Based on how beefy this dependency array is getting, this could be separated into multiple `useEffect`s. I kept it in one since these functions all depend on the state of `loggedIn`, but if anyone feels strongly about this I can split it out.

Reply:
My guess is this is fine, since even though it will run multiple times, it should be quick to run. I would think if you don't see any noticeable performance degradation, it's fine. It's kind of strange that it was written this way to begin with though, not sure why.

Reply:
Yeah I agree, it's interesting how this was written. My guess is to consolidate the `useEffect` calls, although it's debatable whether structuring it like this has any strong advantages. It means this will be called more frequently, but shouldn't be an issue since it's relatively quick like you mentioned. FWIW, the `getProfileInfo` API call is going to be split into smaller separate calls in the near future, so this will almost certainly need to be refactored anyway